pints/_nested/__init__.py

```diff
@@ -780,13 +795,191 @@ def _update_logger(self):
+                    self._logger.log(self._i_message, self._sampler._n_evals,
+                                     self._timer.time(), self._diff,
+                                     float(self._sampler._accept_count /
+                                           (self._sampler._n_evals -
+                                            self._sampler._n_active_points)))
+                else:
+                    self._logger.log(self._i_message, self._sampler._n_evals,
+                                     self._timer.time(), self._diff,
+                                     float(self._sampler._accept_count /
+                                           (self._sampler._n_evals -
+                                            self._sampler._n_active_points)),
+                                     self._sampler._ellipsoid_count)
 
                 # Choose next logging point
                 if self._i_message > self._message_warm_up:
                     self._next_message = self._message_interval * (
                         1 + self._i_message // self._message_interval)
+
+
+class Ellipsoid():
+    """
+    A class to represent N dimensional ellipsoids, which are needed by both
+    ellipsoidal nested sampling and MultiNest.
+
+    In "center form" the equation of an ellipsoid is given by:
+
+    ``(x-c).T * A * (x-c) = 1``
+
+    where ``A`` is a NxN dimensional positive definite symmetric matrix and
+    ``c`` is a N dimensional vector indicating the center of the ellipsoid.
+
+    Parameters
+    ----------
+    A : NxN dimensional positive definite symmetric matrix
+        represents the orientation and size of ellipsoid
+    c : N dimensional vector
+        center of ellipsoid
+    """
+    def __init__(self, A, c):
+
+        self._c = c
+        self._n_parameters = len(c)
+
+        if A.shape != (self._n_parameters, self._n_parameters):
+            raise ValueError(
+                'Sigma must have same dimension as mean, or be a square ' +
+                'matrix with the same dimension as the center.')
+        self._A = np.copy(A)
+
+        # calculate useful quantities
+        self._A_inv = np.linalg.inv(A)
+        # don't calculate volume unless needed
+        self._volume = None
+
+        # don't cache points unless constructed using minimum_volume_ellipsoid
+        self._points = None
+        self._n_points = 0
+
+    def __add__(self, other):
+        return [self, other]
+
+    def add_points(self, points):
+        """ Adds points contained within bounding ellispoid. """
+        self._points = points
+        self._n_points = len(points)
+
+    def centroid(self):
+        """ Returns centroid of ellispoid. """
+        return self._c
+
+    def enlarge(self, enlargement_factor):
+        """ Enlarges ellipsoid by a factor. """
+        self._A *= (1 / enlargement_factor)
+        self._A_inv *= enlargement_factor
+        self._volume = None
+
+    @staticmethod
```

Comment:
Why is this not just a method mahalanobis_distance(self, point, c) ?
Or even mahalanobis_distance(self, point), with c replaced by self._c in the equation?

Reply:
I could have a separate method, that's fine. Just thought since it'd only be used within ellipsoid stuff, I would put it here.

I don't want to have the later `mahalanobis_distance(self, point)` because I want to have the freedom to use the method without actually having to construct ellipsoid objects to avoid the overhead (as is done in some of the functions that use it).